worker/meson.build

```diff
@@ -0,0 +1,227 @@
+  'src/RTC/RateCalculator.cpp',
+  'src/RTC/Router.cpp',
+  'src/RTC/RtpListener.cpp',
+  'src/RTC/RtpObserver.cpp',
+  'src/RTC/RtpPacket.cpp',
+  'src/RTC/RtpProbationGenerator.cpp',
+  'src/RTC/RtpStream.cpp',
+  'src/RTC/RtpStreamRecv.cpp',
+  'src/RTC/RtpStreamSend.cpp',
+  'src/RTC/RtxStream.cpp',
+  'src/RTC/SctpAssociation.cpp',
+  'src/RTC/SctpListener.cpp',
+  'src/RTC/SenderBandwidthEstimator.cpp',
+  'src/RTC/SeqManager.cpp',
+  'src/RTC/SimpleConsumer.cpp',
+  'src/RTC/SimulcastConsumer.cpp',
+  'src/RTC/SrtpSession.cpp',
+  'src/RTC/StunPacket.cpp',
+  'src/RTC/SvcConsumer.cpp',
+  'src/RTC/TcpConnection.cpp',
+  'src/RTC/TcpServer.cpp',
+  'src/RTC/Transport.cpp',
+  'src/RTC/TransportCongestionControlClient.cpp',
+  'src/RTC/TransportCongestionControlServer.cpp',
+  'src/RTC/TransportTuple.cpp',
+  'src/RTC/TrendCalculator.cpp',
+  'src/RTC/UdpSocket.cpp',
+  'src/RTC/WebRtcTransport.cpp',
+  'src/RTC/Codecs/H264.cpp',
+  'src/RTC/Codecs/VP8.cpp',
+  'src/RTC/Codecs/VP9.cpp',
+  'src/RTC/RtpDictionaries/Media.cpp',
+  'src/RTC/RtpDictionaries/Parameters.cpp',
+  'src/RTC/RtpDictionaries/RtcpFeedback.cpp',
+  'src/RTC/RtpDictionaries/RtcpParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpCodecMimeType.cpp',
+  'src/RTC/RtpDictionaries/RtpCodecParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpEncodingParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpHeaderExtensionParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpHeaderExtensionUri.cpp',
+  'src/RTC/RtpDictionaries/RtpParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpRtxParameters.cpp',
+  'src/RTC/SctpDictionaries/SctpStreamParameters.cpp',
+  'src/RTC/RTCP/Packet.cpp',
+  'src/RTC/RTCP/CompoundPacket.cpp',
+  'src/RTC/RTCP/SenderReport.cpp',
+  'src/RTC/RTCP/ReceiverReport.cpp',
+  'src/RTC/RTCP/Sdes.cpp',
+  'src/RTC/RTCP/Bye.cpp',
+  'src/RTC/RTCP/Feedback.cpp',
+  'src/RTC/RTCP/FeedbackPs.cpp',
+  'src/RTC/RTCP/FeedbackRtp.cpp',
+  'src/RTC/RTCP/FeedbackRtpNack.cpp',
+  'src/RTC/RTCP/FeedbackRtpTmmb.cpp',
+  'src/RTC/RTCP/FeedbackRtpSrReq.cpp',
+  'src/RTC/RTCP/FeedbackRtpTllei.cpp',
+  'src/RTC/RTCP/FeedbackRtpEcn.cpp',
+  'src/RTC/RTCP/FeedbackRtpTransport.cpp',
+  'src/RTC/RTCP/FeedbackPsPli.cpp',
+  'src/RTC/RTCP/FeedbackPsSli.cpp',
+  'src/RTC/RTCP/FeedbackPsRpsi.cpp',
+  'src/RTC/RTCP/FeedbackPsFir.cpp',
+  'src/RTC/RTCP/FeedbackPsTst.cpp',
+  'src/RTC/RTCP/FeedbackPsVbcm.cpp',
+  'src/RTC/RTCP/FeedbackPsLei.cpp',
+  'src/RTC/RTCP/FeedbackPsAfb.cpp',
+  'src/RTC/RTCP/FeedbackPsRemb.cpp',
+  'src/RTC/RTCP/XR.cpp',
+  'src/RTC/RTCP/XrDelaySinceLastRr.cpp',
+  'src/RTC/RTCP/XrReceiverReferenceTime.cpp',
+]
+
+cpp = meson.get_compiler('cpp')
+
+openssl_dep = dependency(
+  'openssl',
+  static: get_option('default_library') == 'static',
+  version: '>= 1.1.1'
+)
+
+if cpp.has_header('nlohmann/json.hpp')
+  nlohmann_json_dep = declare_dependency()
+else
+  nlohmann_json_proj = subproject('nlohmann_json')
+  nlohmann_json_dep = nlohmann_json_proj.get_variable('nlohmann_json_dep')
```

Comment:
I'd just like to note, you do not need this...

Simply using `dependency('nlohmann_json')` will first check for nlohmann_json on the system using pkg-config (installed in upstream nlohmann_json's meson.build), then check for it using cmake (installed in upstream nlohmann_json's CMakeLists.txt), then finally check for it via the wrap file which specifies:

```
[provide]
nlohmann_json = nlohmann_json_dep
```

Reply:
I wasn't sure how it is called in an OS, this is the first time I use Meson, thanks for suggestion!